### PR TITLE
GenAI - Refactor Service organization - Allow for mixing LLMs

### PR DESCRIPTION
**Pull request by dandrabik**
## WHAT
I reorganized a bunch of file structure since a clear pattern emerged with the 3 LLM prompts (lots of files/folders with prefixes).

There's also a small code change to add the ability to use different APIS for each step, e.g. Gemini for primary and OpenAI for Repeated and Secondary and a new prompt.
## WHY
I think this organization is more simple and less confusing. I'm adding another piece to this (in a later PR) and it started to feel messy.
## HOW
Organize into 3 main folders `primary_feedback`, `secondary_feedback`, and `repeated_feedback`. Screenshots are the easiest way to parse this.

Note, the diff implies more changes than were there since I had to move files AND add a module line, e.g. `module SecondaryFeedback` so it appears as an entire 'new code file' in the diff.
### Screenshots
**Before**
Screenshot 2024-08-21 at 11 20 27 AM
**After**
Screenshot 2024-08-21 at 11 19 44 AM
**Within a folder**
Screenshot 2024-08-21 at 11 19 54 AM

### What have you done to QA this feature?
This is almost purely a refactor, so mostly going on tests and running the local `thor` tasks

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  Yes
Have you deployed to Staging? |  NO - non-app change
Self-Review: Have you done an initial self-review of the code below on Github? | Yes

### CODE REVIEW

File: services/QuillLMS/engines/evidence/lib/evidence/check/gen_ai.rb [hunk 1]
```diff
@@ -3,8 +3,9 @@
 module Evidence
   module Check
     class GenAI < Check::Base
-      CHAT_API = Evidence::Gemini::Chat
-      SMALL_MODEL = CHAT_API::SMALL_MODEL
+      FEEDBACK_API = Evidence::Gemini::Chat
+      REPEAT_API = Evidence::OpenAI::Chat
+      SECONDARY_API = Evidence::OpenAI::Chat
```
Comment (dandrabik) [inline services/QuillLMS/engines/evidence/lib/evidence/check/gen_ai.rb:8]: This is the one functionality change in the PR.

File: services/QuillLMS/engines/evidence/app/services/evidence/gen_ai/repeated_feedback/checker.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+          @feedback = feedback
+          @history = history
+          @optimal = optimal
+          @chat_api = chat_api
+        end
+
+        def run
+          return false if optimal
+          return false if history.empty? # can't be repeat with no history
+          return false if history.size >= 4 # No secondary feedback on last attempt
```
Comment (brendanshean) [inline services/QuillLMS/engines/evidence/app/services/evidence/gen_ai/repeated_feedback/checker.rb]: Maybe extract a constant: `PENULTIMATE_ATTEMPT = 4` and get rid of comment?

**Review by brendanshean — APPROVED** [review]
I think this reorganization is much cleaner.

File: services/QuillLMS/engines/evidence/app/services/evidence/gen_ai/repeated_feedback/checker.rb (hunk 2 above)
Comment (dandrabik) [reply services/QuillLMS/engines/evidence/app/services/evidence/gen_ai/repeated_feedback/checker.rb]: Good feedback. I've refactored this to make the code a little more obvious (and the comment something that the code doesn't give).

**Pull request merged**